heron/api/src/java/com/twitter/heron/api/Config.java

```diff
@@ -147,6 +147,31 @@
    */
   public static final String TOPOLOGY_COMPONENT_RAMMAP = "topology.component.rammap";
   /**
+   * Is this topology stateful? The format of this flag is boolean
+   */
+  public static final String TOPOLOGY_STATEFUL = "topology.stateful";
+  /**
+   * What's the checkpoint interval for stateful topologies in seconds
+   */
+  public static final String TOPOLOGY_STATEFUL_CHECKPOINT_INTERVAL =
+                             "topology.stateful.checkpoint.interval";
+  /**
+   * What's the provider for state? i.e. one where state is stored
+   */
+  public static final String TOPOLOGY_STATEFUL_PROVIDER_TYPE =
```

Comment:
Is this the implementation class? If so can we use `class` instead of type to be more explicit?